### PR TITLE
Troubleshooting tips for cross-account admin perms and "Microsoft Edge can't read or write to its data directory" error

### PR DESCRIPTION
**Pull request by AlbinoDrought**
# Description

https://discord.com/channels/1042734330029547630/1154062298776862750

![image](https://github.com/wailsapp/wails/assets/852873/60b71cb4-ce5a-4a90-a6e8-dca2d0c3b8e7)

Troubleshooting tips for cross-account admin perms and "Microsoft Edge can't read or write to its data directory" error. Hopefully useful for at least Google fodder :wink: 

## Type of change
  
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Docs only

# How Has This Been Tested?

- [x] Windows
- [ ] macOS
- [ ] Linux

None of the possible solutions are perfect. In my opinion, the in-memory solution would be the best but it would need to be implemented in upstream WebView2 (["fairly low on [their] backlog"](https://github.com/MicrosoftEdge/WebView2Feedback/issues/3637#issuecomment-1668668484))

I use possible solution 1 in my application. It is not without faults. One fault: if you launch an elevated updater, and that updater launches your application, by default your application will be started with the elevated permissions and you will run into the same issue again. 


<details>
<summary>other stuff</summary>

## Test Configuration

```

                                
          Wails Doctor          
                                

                                                                                                                                                                                                              
# Wails
Version         | v2.6.0
Package Manager | pacman

# System
┌──────────────────────────────┐
| OS           | Manjaro Linux |
| Version      | Unknown       |
| ID           | manjaro       |
| Go Version   | go1.21.0      |
| Platform     | linux         |
| Architecture | amd64         |
└──────────────────────────────┘

# Dependencies
┌─────────────────────────────────────────────────────┐
| Dependency | Package Name | Status    | Version     |
| *docker    | docker       | Installed | 1:24.0.5-1  |
| gcc        | gcc          | Installed | 13.2.1-3    |
| libgtk-3   | gtk3         | Installed | 1:3.24.38-1 |
| libwebkit  | webkit2gtk   | Installed | 2.40.5-2    |
| npm        | npm          | Installed | 10.1.0-1    |
| pkg-config | pkgconf      | Installed | 1.8.1-1     |
└────────────── * - Optional Dependency ──────────────┘

# Diagnosis
 SUCCESS  Your system is ready for Wails development!

 ♥   If Wails is useful to you or your company, please consider sponsoring the project:
https://github.com/sponsors/leaanthony
```

# Checklist:

- [ ] I have updated `website/src/pages/changelog.mdx` with details of this PR
- [ ] My code follows the general coding style of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

</details>

### CODE REVIEW

File: website/docs/guides/troubleshooting.mdx [hunk 1]
```diff
@@ -203,9 +203,61 @@ you can make the webview background transparent using the following config:
...
+
+    err := windows.ShellExecute(0, verbPtr, exePtr, argPtr, cwdPtr, showCmd)
+    if err != nil {
+        return err
+    }
+    return nil
+}
+```
+
+Possible solution #2: change `WebviewUserDataPath` to a path writeable by both Alice and Bob.
```
Comment (dkajtazi) [inline website/docs/guides/troubleshooting.mdx:254]: Tried this but it didn't work. Error: https://user-images.githubusercontent.com/93655205/268789148-358c596b-dd25-4207-8d20-52b8c66ca281.png

Comment (AlbinoDrought) [reply website/docs/guides/troubleshooting.mdx:254]: Interesting. I took time to setup a VM and can confirm that it also doesn't work for me, even if using a non-userprofile directory ACL'd to everyone.

However, if I use an elevated Git bash to launch the program (still using the ACL'd directory), all the child Webview2 processes launch under Bob's account and the application works :thinking: 

image

I'm using the default template with this main.go:

```go
package main

import (
	"embed"
	"os"

	"github.com/hectane/go-acl"
	"github.com/wailsapp/wails/v2"
	"github.com/wailsapp/wails/v2/pkg/options"
	"github.com/wailsapp/wails/v2/pkg/options/assetserver"
	"github.com/wailsapp/wails/v2/pkg/options/windows"
)

//go:embed all:frontend/dist
var assets embed.FS

func main() {
	var windowsOptions *windows.Options
	if len(os.Args) > 1 {
		dir := os.Args[1]
		println("Setting data dir to", dir)
		if err := os.MkdirAll(dir, os.ModePerm); err != nil {
			println("Failed creating dir:", err)
		}
		if err := acl.Chmod(dir, 0777); err != nil {
			println("Failed setting ACL on dir:", err)
		}
		windowsOptions = &windows.Options{
			WebviewUserDataPath: dir,
		}
	}

	// Create an instance of the app structure
	app := NewApp()

	// Create application with options
	err := wails.Run(&options.App{
		Title:  "sample-data-dir",
		Width:  1024,
		Height: 768,
		AssetServer: &assetserver.Options{
			Assets: assets,
		},
		BackgroundColour: &options.RGBA{R: 27, G: 38, B: 54, A: 1},
		OnStartup:        app.startup,
		Bind: []interface{}{
			app,
		},
		Windows: windowsOptions,
	})

	if err != nil {
		println("Error:", err.Error())
	}
}
```

Comment (AlbinoDrought) [reply website/docs/guides/troubleshooting.mdx:254]: Hello,

As a disclaimer, let it be known that I have understanding of the Windows security model.

The tokens used when running the program with Git Bash seem to have more permissions, allowing it to work without receiving a `80010108 The object invoked has disconnected from its clients` error:

image

Left is the binary running regularly. Right is the binary running from an elevated Git Bash.

I made a PoC Wails app that re-runs itself using a token with the same permissions. It works in the above Bob/Alice split-account scenario: https://github.com/AlbinoDrought/sample-split-account-admin-wails/commit/823a03de826c7e1948fbda7e44b30083f774c9a0

Does this work for you as well?

Cheers

Comment (AlbinoDrought) [reply website/docs/guides/troubleshooting.mdx]: Added to troubleshooting doc in https://github.com/wailsapp/wails/compare/4c44b6d66b0029a69818d0b3f2bdb4a840f7279f..afc1750ef6aadde2713c7150d7d11a0b68ac008d

**Comment by leaanthony** [comment]
Thanks @AlbinoDrought 👍

**Pull request merged**